### PR TITLE
sys/architecture: add UWORD_MIN, UWORD_MAX, SWORD_MIN, SWORD_MAX

### PR DESCRIPTION
**Pull request by maribu**
### Contribution description

This PR adds preprocessor macros for `UWORD_MIN`, `UWORD_MAX`, `SWORD_MIN`, `SWORD_MAX`. The related to `uword_t` and `sword_t`, respectively, the same way as `UINT8_MAX` and `UINT8_MIN` behaves to `uint8_t`.

### Testing procedure

The test application has been extended. As those are covered with a `static_assert()`, a successful compilation already counts as pass.

### Issues/PRs references

None

### CODE REVIEW

File: sys/include/architecture.h [hunk 1]
```diff
@@ -87,6 +87,26 @@ typedef int32_t     sword_t;
  */
 #define WORD_ALIGNED __attribute__((aligned(ARCHITECTURE_WORD_BYTES)))
 
+/**
+ * @brief   Smallest number an uwort_t can hold
```
Comment (kaspar030) [inline sys/include/architecture.h:91]:
```suggestion
 * @brief   Smallest number an uword_t can hold
```

File: sys/include/architecture.h [hunk 2]
```diff
@@ -87,6 +87,26 @@ typedef int32_t     sword_t;
...
+ */
+#define UWORD_MIN                   0
+
+/**
+ * @brief   Highest number an uword_t can hold
+ */
+#define UWORD_MAX                   ((1ULL << ARCHITECTURE_WORD_BITS) - 1)
+
+/**
+ * @brief   Smallest number an sword_t can hold
```
Comment (kaspar030) [inline sys/include/architecture.h:101]:
```suggestion
 * @brief   Smallest number a sword_t can hold
```

File: sys/include/architecture.h [hunk 3]
```diff
@@ -87,6 +87,26 @@ typedef int32_t     sword_t;
...
+ */
+#define SWORD_MIN                   (-(1LL << (ARCHITECTURE_WORD_BITS - 1)))
+
+/**
+ * @brief   Highest number an sword_t can hold
```
Comment (kaspar030) [inline sys/include/architecture.h:106]:
```suggestion
 * @brief   Highest number a sword_t can hold
```

File: sys/include/architecture.h (hunk 2 above)
Comment (maribu) [reply sys/include/architecture.h:101]: "an" is actually right here, because you pronounce `sword_t` as "es-word-tea`

**Review by kaspar030 — APPROVED** [review]
ACK.

File: sys/include/architecture.h (hunk 1 above)
Comment (maribu) [reply sys/include/architecture.h]: fixed

**Comment by miri64** [comment]
:crossed_swords:

File: sys/include/architecture.h (hunk 2 above)
Comment (bergzand) [inline sys/include/architecture.h:101]:
```suggestion
 * @brief   Smallest number an sword_t (pronounced 's-word') can hold
```

:stuck_out_tongue:

Comment (maribu) [reply sys/include/architecture.h:104]: Solved by https://github.com/RIOT-OS/RIOT/pull/15631/commits/a4b011f6fc661b5abb751fed615f095354108a27 ?

File: sys/include/architecture.h [hunk 4]
```diff
@@ -58,6 +58,9 @@ typedef uint<num>_t uword_t;
  *
  * @details Synonym to `int8_t`, `int16_t` or `int32_t` depending on
  *          architecture
+ *
+ * @note    This type is pronounce es-word-tea. When slaying dragons, this is
+ *          not the tool you're looking for.
```
Comment (miri64) [inline sys/include/architecture.h:63]: :dragon: :princess: :crossed_swords: :disappointed:

**Pull request merged**